src/main/java/cs361/battleships/models/Board.java

```diff
@@ -36,6 +37,131 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		} else return false;
 	}
 
+	public void moveShips(String direction)
+	{
+		List<Ship> sortedShips = new ArrayList<>();
+		Ship curShip = getDirectionmostShip(direction);
+		var ships = this.getShips();
+		while(curShip != null)
+		{
+			sortedShips.add(curShip);
+			ships.remove(curShip);
+			curShip = getDirectionmostShip(direction);
+		}
+		this.ships = sortedShips;
+
+		for(var ship : this.getShips())
+		{
+			System.out.println(ship.getKind());
+			var proposedMove = ship.getMove(direction);
+			if(validMove(proposedMove, ship, this.getShips()))
+			{
+				ship.move(direction);
+			}
+			else
+			{
+				System.out.println("FALSE");
+			}
+		}
+	}
+
+	public boolean validMove(List<Square> proposedMove, Ship proposedShip, List<Ship> ships)
+	{
+		for(var square : proposedMove)
+		{
+			if(square.getColumn() > 'J' || square.getColumn() < 'A')
+			{
+				System.out.println("FAILING COLUMN CHECK");
+				return false;
+			}
+			if(square.getRow() > 10 || square.getRow() < 1)
+			{
+				System.out.println("FAILING ROW CHECK");
+				return false;
+			}
+		}
+
+		for(var propSq : proposedMove)
+		{
+			//loop through every ship
+			for(var ship : ships)
+			{
+				//loop through every square in the ship
+				for(var sq : ship.getOccupiedSquares())
+				{
+					//compare each square to the proposed square.
+					if(propSq.getRow() == sq.getRow() && propSq.getColumn() == sq.getColumn())
+					{
+						if(!proposedShip.equals(ship) && !ship.submerged && !proposedShip.submerged)
+						{
+							return false;
+						}
+					}
+				}
+			}
+		}
+
+
+		return true;
+	}
+
+	public Ship getDirectionmostShip(String direction)
+	{
+		if(this.getShips().size() == 0)
```

Comment:
I think stylistically I like it more condensed, but we don't really have an agreed upon style actually. Also I noticed comments are missing here. I do think it reads just fine regardless, but I wonder if there was a more concise way to do the same thing (maybe not).